test/agentchat/test_groupchat.py

```diff
@@ -49,6 +51,181 @@ def test_func_call_groupchat():
     agent2.initiate_chat(group_chat_manager, message={"function_call": {"name": "func", "arguments": '{"x": 1}'}})
 
 
+def test_group_chat_math_class():
+    """
+    This test case is to simulate a math class.
+    where teacher creates math questions and student resolves the questions.
+    teacher will create a question, student will resolve the question and tell teacher the answer.
+    If the answer is correct, teacher will create another question, otherwise, teacher will ask student to resolve the question again.
+    The class will end when teacher has created 3 questions.
+
+    This test case is created to test the following features:
+    - speaker selection should work under a continuous q&a scenario among two agents and GPT 3.5 model.
+    - admin should end the class when teacher has created 3 questions.
+    """
+    skip_if_openai_not_available()
```

Comment:
Nice. This is better than my old way. @rickyloynd-microsoft @thinkall @kevin666aa FYI.